### ISSUE TITLE
New model request: Mixed effect random forest (MERF)

### ISSUE THREAD

**Issue by amcmahon17** [tidymodels/multilevelmod#50]
Hello, I have a new model request: Mixed effect random forest (MERF). There are a few packages that support them in R. I'm not sure if this is the proper package for it.

**Comment by hfrick** [comment]
Hi @amcmahon17 ! Could you say a bit more about your use case for them and point us to the most suitable implementations in R and references?

**Comment by amcmahon17** [comment]
Hello! Thank you for the response. Please let me know if the following is
sufficient.

The use case is repeated measures/longitudinal data and clustered data.
It's basically the same as ordinary mixed effects. MERFs have been shown to
have a performance boost vs ordinary random forests.

The best package I have found is LongituRF. The relevant function is
MERF().

I have references for two papers below. The second is by the author of the
LongituRF package.

-https://cran.r-project.org/web/packages/LongituRF/index.html
-"Mixed-effects random forest for clustered data"
https://www.tandfonline.com/doi/abs/10.1080/00949655.2012.741599
 (Journal of Statistical Computation and Simulation
Volume 84, 2014 - Issue 6)

-"Random forests for high-dimensional longitudinal data"
https://journals.sagepub.com/doi/10.1177/0962280220946080
(Statistical Methods in Medical Research Volume 30, Issue 1)

-"Comparing (Stochastic) Mixed-Effects Tree-Based Models in {LongituRF} &
{ranger}", Blog of Ph.D. student Christopher Loan:
https://www.christopherloan.com/blog/running-longitudinal-random-forests-with-longiturf/

**Comment by amcmahon17** [comment]
Sorry, my references didn't come through for some reason. Here they are:

LongituRF package.

-https://cran.r-project.org/web/packages/LongituRF/index.html
-"Mixed-effects random forest for clustered data"
https://www.tandfonline.com/doi/abs/10.1080/00949655.2012.741599
 (Journal of Statistical Computation and Simulation
Volume 84, 2014 - Issue 6)

-"Random forests for high-dimensional longitudinal data"
https://journals.sagepub.com/doi/10.1177/0962280220946080
(Statistical Methods in Medical Research Volume 30, Issue 1)


-"Comparing (Stochastic) Mixed-Effects Tree-Based Models in {LongituRF} & {ranger}", Blog of Ph.D. student Christopher Loan: https://www.christopherloan.com/blog/running-longitudinal-random-forests-with-longiturf/

**Comment by hfrick** [comment]
Thank you! I'll give this a closer look the next time we are returning to more multilevelmod development 🪆

**Comment by jscamac** [comment]
This is a really nicely written paper that summarises the use-case and some of the R packages available in R.
https://academic.oup.com/bib/article/24/2/bbad002/6991123

**Comment by marianawebb** [comment]
Agreed that it would be awesome to have MERF (or something similar like MixRF) in tidymodels. I'll be following

**Comment by sebsilas** [comment]
Anyone been working on this? It would be great to have!

**Comment by jprice80** [comment]
I agree that this would be great to add.  Longitudinal use cases are increasingly more common.  Consequently, this would be great to add as less intensive alternatives compared to LSTM keras models.

**Comment by UrielMenalled** [comment]
Agreed! MERFs would be great to have in TidyModels!
